### DISCUSSION TITLE
0.3.0-rc1 has been released!

### DISCUSSION THREAD

**Discussion by mihe** [godot-jolt/godot-jolt#465] [Announcements]
This is the first **release candidate** for version 0.3.0, and assuming no major/critical bugs or regressions are found is likely to become 0.3.0-stable.

This release supports [**Godot 4.0.3**](https://github.com/godotengine/godot/releases/tag/4.0.3-stable).

## Known major issues

- Extension classes cause exception when used from C# ([#270](https://github.com/godot-jolt/godot-jolt/issues/270))
- The `move_and_collide` parameter `recovery_as_collision` is always `true` ([#337](https://github.com/godot-jolt/godot-jolt/issues/337))

## Changelog

These are the notable changes that have been made since [0.2.3-stable](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.2.3-stable) was released. You can also find a list of all the commits [here](https://github.com/godot-jolt/godot-jolt/compare/v0.2.3-stable...v0.3.0-rc1).

Any breaking changes are denoted with ⚠️.

### Changed

- ⚠️ Changed collision layers and masks to behave as they do in Godot Physics, allowing for asymmetrical collisions, where the body whose mask does not contain the layer of the other body effectively gets infinite mass and inertia in the context of that collision.

### Added

- Added new project setting, "Use Shape Margins", which when disabled leads to all shape margins being ignored and instead set to 0, at a slight performance cost.
- Added new project setting, "Areas Detect Static Bodies", to allow `Area3D` to detect overlaps with static bodies (including `RigidBody3D` using `FREEZE_MODE_STATIC`) at a potentially [heavy performance/memory cost](https://github.com/godot-jolt/godot-jolt/blob/master/docs/settings.md#jolt-3d).

### Fixed

- Fixed issue where a `RigidBody3D` using `FREEZE_MODE_KINEMATIC` wouldn't have its `_integrate_forces` method called when monitoring contacts.
- Fixed issue where scaling bodies/shapes with negative values would break them in various ways.
- Fixed issue where `CharacterBody3D` platform velocities would always be zero.
- Fixed issue where the velocity of kinematic colliders would always be zero in `_physics_process`.

<hr /><em>This discussion was created from the release <a href='https://github.com/godot-jolt/godot-jolt/releases/tag/v0.3.0-rc1'>0.3.0-rc1</a>.</em>

**Comment by mgiraud** [comment]
The "Changed collision layers and mask" has quite changed the physics of my pinball. I have to be smarter on my layers and mask choices. Playground vs ball and flippers vs ball, i hope to find some good params for a better simulation. And i would have never thought that making a pinball would be so hard on the physic engine part ^^
Good job ! Can't wait for joints new features ^^

**Reply by mihe** [reply]
> The "Changed collision layers and mask" has quite changed the physics of my pinball.

The difference in behavior can indeed be quite significant, depending on your setup, so I'm very thankful for the generous support from @jrouwe that allowed this to be brought in line with how Godot Physics behaves. It should hopefully make any transition to/from Godot Physics even smoother.

(And of course, if it does not in fact behave like Godot Physics, then please let me know.)
